objects/src/block/header.rs

```diff
@@ -275,6 +275,39 @@ impl Deserializable for BlockHeader {
     }
 }
 
+
+
+/// BLOCK NUMBER STRUCT
+
+/// Holds `u32` type  to signify Block Number
+
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone, PartialOrd, Ord)]
+pub struct BlockNumber(u32);
+
+impl Serializable for BlockNumber {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u32(self.0);
+    }
+
+    fn get_size_hint(&self) -> usize {
+        core::mem::size_of::<u32>()
+    }
+}
+
+impl From<u32> for BlockNumber {
+    fn from(value: u32) -> Self {
+        BlockNumber(value)
+    }
+}
+
+impl BlockNumber{
```

Comment:
We should also add a constructor from an epoch here, so that we can replace this:

https://github.com/0xPolygonMiden/miden-base/blob/c98b1b7b3c78ad9ae4c8f690763fa15a4c22838d/objects/src/transaction/inputs.rs#L480

with

```rust
let anchor_block_number = BlockNumber::from_epoch(account.id().anchor_epoch());
```

It should then be possible to completely remove `block_num_from_epoch`.